### ISSUE TITLE
type with local variable removes constructor

### ISSUE THREAD

**Issue by filmackay** [JuliaLang/julia#5790]
```julia
julia> type Test
           a::Int
           b::Int = 0 # this seems to remove the constructor
       end

julia> Test
Test (constructor with 0 methods)

julia> Test.names, Test.types
((:a,),(Int64,))
```

**Comment by mschauer** [comment]
Close. This issue is closed by 7a4aa3e and superseded by #10146

**Comment by ivarne** [comment]
Correct! Thanks @mschauer
